client/state/experiments/hooks.ts

```diff
@@ -1,60 +1,68 @@
 /**
  * External Dependencies
  */
-import { useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal Dependencies
  */
-import { getVariationForUser, isLoading, getAnonId, nextRefresh } from './selectors';
+import { fetchExperiments } from 'calypso/state/data-layer/wpcom/experiments';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { getVariationForUser, isLoading, nextRefresh } from './selectors';
 
 /**
- * Returns true if the variations are loading for the current user
+ * Keeps track of whether an event has fired to generate an anonId
+ * Internal state not to be used anywhere else
  */
-export function useIsLoading(): boolean {
-	return useSelector( isLoading );
-}
+let hasAnonIdGeneratingTracksEventFired = false;
 
 /**
- * Returns the user's assigned variation for a given experiment
+ * Gets (and initialises if not yet initialised) a users assignment to an experiment.
  *
- * @param experiment The name of the experiment
- */
-export function useVariationForUser( experiment: string ): string | null {
-	return useSelector( ( state ) => getVariationForUser( state, experiment ) );
-}
-
-/**
- * Gets the anon id for the user, if set
- */
-export function useAnonId(): string | null {
-	return useSelector( getAnonId );
-}
-
-/**
- * Get the time for the next variation refresh
+ * Usage tips:
+ * - Account for loading state as assignment is now server-side.
+ * - Account for null variations: Provide the fallback experience.
```

Comment:
It'd help to make this more detailed. Maybe add a few examples?

It may be best to add detailed examples to the readme and reference them here.